electron/src/electron-main.js

```diff
@@ -0,0 +1,292 @@
+  var appPath = app.getPath('exe');
+  var iconPath = (() => {
+    switch (process.platform) {
+      case 'linux':
+        return path.join(path.dirname(appPath), './resources/icon512x512.png');
+    }
+  })()
+
+  // Create the browser window.
+  win = new BrowserWindow({
+    width: 1200,
+    height: 900,
+    title: 'Skycoin',
+    icon: iconPath,
+    nodeIntegration: false,
+    webPreferences: {
+      webgl: false,
+      webaudio: false,
+      contextIsolation: true,
+      webviewTag: false,
+      nodeIntegration: false,
+      nodeIntegrationInWorker: false,
+      allowRunningInsecureContent: false,
+      webSecurity: true,
+      plugins: false,
+    },
+  });
+
+  // patch out eval
+  win.eval = global.eval;
+  win.webContents.executeJavaScript('window.eval = 0;');
+
+  win.loadURL(`http://127.0.0.1:` + serverPort);
+
+  const ses = win.webContents.session
+  ses.clearCache(function () {
+    console.log('Cleared the caching of the skycoin wallet.');
+  });
+
+  // Emitted when the window is closed.
+  win.on('closed', () => {
+    // Dereference the window object, usually you would store windows
+    // in an array if your app supports multi windows, this is the time
+    // when you should delete the corresponding element.
+    win = null;
+  });
+
+  win.webContents.on('will-navigate', function(e, url) {
+    e.preventDefault();
+    require('electron').shell.openExternal(url);
+  });
+
+  // create application's main menu
+  var template = [{
+    label: 'Skycoin',
+    submenu: [
+      { label: 'Quit', accelerator: 'Command+Q', click: function() { app.quit(); } }
+    ]
+  }, {
+    label: 'Edit',
+    submenu: [
+      { label: 'Undo', accelerator: 'CmdOrCtrl+Z', selector: 'undo:' },
+      { label: 'Redo', accelerator: 'Shift+CmdOrCtrl+Z', selector: 'redo:' },
+      { type: 'separator' },
+      { label: 'Cut', accelerator: 'CmdOrCtrl+X', selector: 'cut:' },
+      { label: 'Copy', accelerator: 'CmdOrCtrl+C', selector: 'copy:' },
+      { label: 'Paste', accelerator: 'CmdOrCtrl+V', selector: 'paste:' },
+      { label: 'Select All', accelerator: 'CmdOrCtrl+A', selector: 'selectAll:' }
+    ]
+  }, {
+    label: 'Show',
+    submenu: [
+      {
+        label: 'DevTools',
+        accelerator: process.platform === 'darwin' ? 'Alt+Command+I' : 'Ctrl+Shift+I',
+        click: (item, focusedWindow) => {
+          if (focusedWindow) {
+            focusedWindow.toggleDevTools();
+          }
+        }
+      },
+    ]
+  }];
+
+  Menu.setApplicationMenu(Menu.buildFromTemplate(template));
+
+  session
+    .fromPartition('')
+    .setPermissionRequestHandler((webContents, permission, callback) => {
+      return callback(false);
+    });
+
+  // Blocks the connection if the URL is not in allowedHosts.
+  session.defaultSession.webRequest.onBeforeRequest(['*://*./*'], function(details, callback) {
+    if (details.url.startsWith('http')) {
+      if (!nodesAllowed) {
+        allowNodes();
```

Comment:
`allowNodes` does async stuff, it should return the promise so the `callback` can be called when it's done

Reply:
I removed the asynchronous procedure. Now the data is injected synchronously with the Electron ipc